src/Catty/DataModel/UserVariable/UserVariable.swift

```diff
@@ -0,0 +1,71 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import Foundation
+import SpriteKit
+
+@objc(UserVariable)
+
+@objcMembers class UserVariable: NSObject, CBMutableCopying {
+
+    var name: String!
+    var value: Any!
+    var textLabel: SKLabelNode!
+    var isList: Bool
+
+    override init() {
+        self.isList = false
+        self.name = nil
+    }
+
+    init!(variable: UserVariable) {
+        self.name = variable.name
+        self.isList = variable.isList
+    }
+
+    init!(name: String!, isList: Bool) {
+        self.name = name
+        self.isList = isList
+    }
+
+    override var description: String {
+        "UserVariable: Name: \(String(describing: self.name)), Value: \(String(describing: self.value))"
+    }
+
+    func isEqual(toUserVariable userVariable: UserVariable?) -> Bool {
```

Comment:
Could you please try whether you can use the default method `override func isEqual(_ object: Any?) -> Bool`? The name for the ObjectiveC method can still be `isEqualToUserVariable`.

Reply:
That will be helpful, because methods can not be mistaken anymore, like it currently is the case in your changes for the test cases.

Reply:
Overriding that default method isn't resulting in the name `isEqualToUserVariable` for Objective-C. 
Rather, I can use `@objc(isEqualToUserVariable:)` to explicitly mention the method name if that's okay. If I do that, this will be the signature `@objc(isEqualToUserVariable:) func isEqual(to userVariable: UserVariable?) -> Bool`.

Reply:
Moreover, I may also override the default method and call the above method from there.

Reply:
True, because the overridden Objective C method has a different selector. Please then do not use any special name for Objective C -- I would suggest to use the default method `override func isEqual(_ object: Any?) -> Bool` which is perfect for our use case. This will induce a few changes on the existing Objective C classes -- luckily those are not too many (little bit more than 20), so the review will not be too hard.

Please note: it turns out that in some tests (i.e., `ProjectTests`) the `==` operator has been wrongly used to compare equality references (replace it with `===` in those cases). Please verify this by running all tests (except UI tests) locally on your machine.